src/scss/sections/_product-carousel.scss

```diff
@@ -0,0 +1,74 @@
+.product-card {
+
+  &__title {
+    min-height: calc-em(54);
+  }
+
+  .swatch-overlay {
+    width: 100%;
+    height: 100%;
+    display: none;
+    bottom: 0;
+    left: 0;
+    padding-bottom: calc-em(58);
+    background-image: linear-gradient(180deg, rgba(255, 255, 255, 0.9) 0%, white 100%);
+    visibility: visible;
+    transition: all 300ms ease-in-out;
+
+    &:hover {
+      display: flex;
+    }
+  }
+
+  .btn-choose:hover {
+    + .swatch-overlay  {
+      display: flex;
+    }
+  }
+}
+
+.variant-swatch {
+  &__item {
+    &.active {
```

Comment:
please keep hierarchy of styles according to the the times they are used, so first the styles in normal behaviour then when their active/show